Build failure on ubuntu-24.04 when using FetchContent and C++23 - missing includes

# Bug description
I recently started to use `simdutf` in my project, I fetched it using `FetchContent_Declare` as the example showed.
Tried using both `tags/v9.0.0` and `tags/v8.2.0`. I have not tried with previous versions.
While I'm able to build fine locally (on windows using clang 21), my [CI started to complain on Ubuntu](https://github.com/Epixu/Anyness/actions/runs/30086887673/job/89461054997) for a missing definition of `memcpy`.

<details>
<summary>Relevant error log</summary>
<p>
```
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/ascii.h: In function ‘constexpr bool simdutf::scalar::{anonymous}::ascii::validate(InputPtr, size_t)’:
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/ascii.h:25:12: error: ‘memcpy’ is not a member of ‘std’; did you mean ‘wmemcpy’?
   25 |       std::memcpy(&v1, data + pos, sizeof(uint64_t));
      |            ^~~~~~
      |            wmemcpy
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/ascii.h:27:12: error: ‘memcpy’ is not a member of ‘std’; did you mean ‘wmemcpy’?
   27 |       std::memcpy(&v2, data + pos + sizeof(uint64_t), sizeof(uint64_t));
      |            ^~~~~~
      |            wmemcpy
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/ascii.h: In function ‘constexpr simdutf::result simdutf::scalar::{anonymous}::ascii::validate_with_errors(InputPtr, size_t)’:
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/ascii.h:58:12: error: ‘memcpy’ is not a member of ‘std’; did you mean ‘wmemcpy’?
   58 |       std::memcpy(&v1, data + pos, sizeof(uint64_t));
      |            ^~~~~~
      |            wmemcpy
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/ascii.h:60:12: error: ‘memcpy’ is not a member of ‘std’; did you mean ‘wmemcpy’?
   60 |       std::memcpy(&v2, data + pos + sizeof(uint64_t), sizeof(uint64_t));
      |            ^~~~~~
      |            wmemcpy
In file included from /home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/implementation.h:132:
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/atomic_util.h: In function ‘void simdutf::scalar::memcpy_atomic_read(char*, const char*, size_t)’:
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/atomic_util.h:43:10: error: ‘memcpy’ is not a member of ‘std’; did you mean ‘wmemcpy’?
   43 |     std::memcpy(dst, &dst_value, sizeof(uint64_t));
      |          ^~~~~~
      |          wmemcpy
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/atomic_util.h: In function ‘void simdutf::scalar::memcpy_atomic_write(char*, const char*, size_t)’:
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/atomic_util.h:89:10: error: ‘memcpy’ is not a member of ‘std’; did you mean ‘wmemcpy’?
   89 |     std::memcpy(&src_val, src, sizeof(uint64_t)); // Non-atomic read from src
      |          ^~~~~~
      |          wmemcpy
In file included from /home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/implementation.h:136:
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/latin1_to_utf8/latin1_to_utf8.h: In function ‘constexpr size_t simdutf::scalar::{anonymous}::latin1_to_utf8::convert(InputPtr, size_t, OutputPtr)’:
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/latin1_to_utf8/latin1_to_utf8.h:29:11: error: ‘::memcpy’ has not been declared
   29 |         ::memcpy(&v1, data + pos, sizeof(uint64_t));
      |           ^~~~~~
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/latin1_to_utf8/latin1_to_utf8.h:1:1: note: ‘memcpy’ is defined in header ‘<cstring>’; this is probably fixable by adding ‘#include <cstring>’
  +++ |+#include <cstring>
    1 | #ifndef SIMDUTF_LATIN1_TO_UTF8_H
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/latin1_to_utf8/latin1_to_utf8.h:31:11: error: ‘::memcpy’ has not been declared
   31 |         ::memcpy(&v2, data + pos + sizeof(uint64_t), sizeof(uint64_t));
      |           ^~~~~~
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/latin1_to_utf8/latin1_to_utf8.h:31:11: note: ‘memcpy’ is defined in header ‘<cstring>’; this is probably fixable by adding ‘#include <cstring>’
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/latin1_to_utf8/latin1_to_utf8.h: In function ‘size_t simdutf::scalar::{anonymous}::latin1_to_utf8::convert_safe(const char*, size_t, char*, size_t)’:
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/latin1_to_utf8/latin1_to_utf8.h:81:9: error: ‘::memcpy’ has not been declared
   81 |       ::memcpy(&v1, data + pos, sizeof(uint64_t));
      |         ^~~~~~
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/latin1_to_utf8/latin1_to_utf8.h:81:9: note: ‘memcpy’ is defined in header ‘<cstring>’; this is probably fixable by adding ‘#include <cstring>’
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/latin1_to_utf8/latin1_to_utf8.h:83:9: error: ‘::memcpy’ has not been declared
   83 |       ::memcpy(&v2, data + pos + sizeof(uint64_t), sizeof(uint64_t));
      |         ^~~~~~
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/latin1_to_utf8/latin1_to_utf8.h:83:9: note: ‘memcpy’ is defined in header ‘<cstring>’; this is probably fixable by adding ‘#include <cstring>’
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/latin1_to_utf8/latin1_to_utf8.h:90:11: error: ‘::memcpy’ has not been declared
   90 |         ::memcpy(utf8_output + utf8_pos, buf + pos, 16);
      |           ^~~~~~
/home/runner/work/Anyness/Anyness/external/simdutf-src/include/simdutf/scalar/latin1_to_utf8/latin1_to_utf8.h:90:11: note: ‘memcpy’ is defined in header ‘<cstring>’; this is probably fixable by adding ‘#include <cstring>’
```
</p>
</details>

The only place I use simdutf is [here](https://github.com/Epixu/Anyness/blob/8d28e5e74e6715a960126840b4d2710d2bee39ae/source/logger/Logger.cpp#L112).

# Reproduction
1. Fetched the library using CMake:
```cmake
   set(SIMDUTF_TESTS OFF)
   set(SIMDUTF_TOOLS OFF)
   set(SIMDUTF_ICONV OFF)

   FetchContent_Declare(
      simdutf
      GIT_REPOSITORY  https://github.com/simdutf/simdutf.git
      GIT_SHALLOW     TRUE
      GIT_TAG         v9.0.0
      EXCLUDE_FROM_ALL
   )
   FetchContent_MakeAvailable(simdutf)
```
2. Link to my target using
```cmake
target_link_libraries(LangulusAnyness PUBLIC simdutf::simdutf)

// An important detail is probably that I'm using C++23 via:
target_compile_features(LangulusAnyness PUBLIC cxx_std_23)

// CI on ubuntu informs me that "Compiling using the C++ standard:11"
// Locally on windows it says the same.
```
3. Include the header and use the library directly
<details>
<summary>Relevant code snippet</summary>
<p>

```c++
#include <simdutf.h>

//...

void State::Write(const ::std::wstring_view& stdString) const noexcept {
   ::std::string buffer;
   size_t conversion_result = 0;

   if constexpr (sizeof(wchar_t) == 2) {
      auto src = reinterpret_cast<char16_t const*>(stdString.data());
      buffer.resize(simdutf::utf8_length_from_utf16(src, stdString.size()));
      conversion_result = simdutf::convert_utf16_to_utf8_safe(
         src, stdString.size(),
         buffer.data(), buffer.size()
      );
   }
   else if constexpr (sizeof(wchar_t) == 4) {
      auto src = reinterpret_cast<char32_t const*>(stdString.data());
      buffer.resize(simdutf::utf8_length_from_utf32(src, stdString.size()));
      conversion_result = simdutf::convert_utf32_to_utf8(
         src, stdString.size(),
         buffer.data()
      );
   }

   if (not conversion_result)
      Write("<invalid wide string>");
   else
      Write(buffer);
}
```
</p>
</details>


4. Build

# simdutf release
9.0.0 / 8.2.0

# Configuration
The CI fails for the following configurations:
 - OS: ubuntu-24.04 (the stock GitHub runner one)
 - Compilers: both GCC 14.2 and Clang 21 fail, regardless if building with -m64 or -m32. 
 - Optimization setting: seems to be irrelevant, both debug and release builds fail for the same reason
My CI workflow is [here](https://github.com/Epixu/Anyness/blob/wip/.github/workflows/ci.yml) for reference.

I am willing to do a pull request, but I suspect it's just a missing `#include <cstring>` in `portability.h`?

Hope I'm not wasting your time!

--- a/include/simdutf/scalar/ascii.h
+++ b/include/simdutf/scalar/ascii.h
@@ -1,6 +1,8 @@
 #ifndef SIMDUTF_ASCII_H
 #define SIMDUTF_ASCII_H
 
+#include <cstring>
+
 namespace simdutf {
 namespace scalar {
 namespace {

--- a/include/simdutf/scalar/atomic_util.h
+++ b/include/simdutf/scalar/atomic_util.h
@@ -2,6 +2,7 @@
 #define SIMDUTF_ATOMIC_UTIL_H
 #if SIMDUTF_ATOMIC_REF
   #include <atomic>
+  #include <cstring>
 namespace simdutf {
 namespace scalar {
 

--- a/include/simdutf/scalar/latin1_to_utf8/latin1_to_utf8.h
+++ b/include/simdutf/scalar/latin1_to_utf8/latin1_to_utf8.h
@@ -1,6 +1,8 @@
 #ifndef SIMDUTF_LATIN1_TO_UTF8_H
 #define SIMDUTF_LATIN1_TO_UTF8_H
 
+#include <cstring>
+
 namespace simdutf {
 namespace scalar {
 namespace {

--- a/include/simdutf/scalar/utf16_to_utf8/utf16_to_utf8.h
+++ b/include/simdutf/scalar/utf16_to_utf8/utf16_to_utf8.h
@@ -1,6 +1,8 @@
 #ifndef SIMDUTF_UTF16_TO_UTF8_H
 #define SIMDUTF_UTF16_TO_UTF8_H
 
+#include <cstring>
+
 namespace simdutf {
 namespace scalar {
 namespace {

--- a/include/simdutf/scalar/utf16_to_utf8/valid_utf16_to_utf8.h
+++ b/include/simdutf/scalar/utf16_to_utf8/valid_utf16_to_utf8.h
@@ -1,6 +1,8 @@
 #ifndef SIMDUTF_VALID_UTF16_TO_UTF8_H
 #define SIMDUTF_VALID_UTF16_TO_UTF8_H
 
+#include <cstring>
+
 namespace simdutf {
 namespace scalar {
 namespace {

--- a/include/simdutf/scalar/utf32_to_latin1/utf32_to_latin1.h
+++ b/include/simdutf/scalar/utf32_to_latin1/utf32_to_latin1.h
@@ -1,6 +1,8 @@
 #ifndef SIMDUTF_UTF32_TO_LATIN1_H
 #define SIMDUTF_UTF32_TO_LATIN1_H
 
+#include <cstring>
+
 namespace simdutf {
 namespace scalar {
 namespace {

--- a/include/simdutf/scalar/utf32_to_latin1/valid_utf32_to_latin1.h
+++ b/include/simdutf/scalar/utf32_to_latin1/valid_utf32_to_latin1.h
@@ -1,6 +1,8 @@
 #ifndef SIMDUTF_VALID_UTF32_TO_LATIN1_H
 #define SIMDUTF_VALID_UTF32_TO_LATIN1_H
 
+#include <cstring>
+
 namespace simdutf {
 namespace scalar {
 namespace {

--- a/include/simdutf/scalar/utf32_to_utf8/utf32_to_utf8.h
+++ b/include/simdutf/scalar/utf32_to_utf8/utf32_to_utf8.h
@@ -1,6 +1,8 @@
 #ifndef SIMDUTF_UTF32_TO_UTF8_H
 #define SIMDUTF_UTF32_TO_UTF8_H
 
+#include <cstring>
+
 namespace simdutf {
 namespace scalar {
 namespace {

--- a/include/simdutf/scalar/utf32_to_utf8/valid_utf32_to_utf8.h
+++ b/include/simdutf/scalar/utf32_to_utf8/valid_utf32_to_utf8.h
@@ -1,6 +1,8 @@
 #ifndef SIMDUTF_VALID_UTF32_TO_UTF8_H
 #define SIMDUTF_VALID_UTF32_TO_UTF8_H
 
+#include <cstring>
+
 namespace simdutf {
 namespace scalar {
 namespace {

--- a/include/simdutf/scalar/utf8.h
+++ b/include/simdutf/scalar/utf8.h
@@ -1,6 +1,8 @@
 #ifndef SIMDUTF_UTF8_H
 #define SIMDUTF_UTF8_H
 
+#include <cstring>
+
 namespace simdutf {
 namespace scalar {
 namespace {

--- a/include/simdutf/scalar/utf8_to_latin1/utf8_to_latin1.h
+++ b/include/simdutf/scalar/utf8_to_latin1/utf8_to_latin1.h
@@ -1,6 +1,8 @@
 #ifndef SIMDUTF_UTF8_TO_LATIN1_H
 #define SIMDUTF_UTF8_TO_LATIN1_H
 
+#include <cstring>
+
 namespace simdutf {
 namespace scalar {
 namespace {

--- a/include/simdutf/scalar/utf8_to_latin1/valid_utf8_to_latin1.h
+++ b/include/simdutf/scalar/utf8_to_latin1/valid_utf8_to_latin1.h
@@ -1,6 +1,8 @@
 #ifndef SIMDUTF_VALID_UTF8_TO_LATIN1_H
 #define SIMDUTF_VALID_UTF8_TO_LATIN1_H
 
+#include <cstring>
+
 namespace simdutf {
 namespace scalar {
 namespace {

--- a/include/simdutf/scalar/utf8_to_utf16/utf8_to_utf16.h
+++ b/include/simdutf/scalar/utf8_to_utf16/utf8_to_utf16.h
@@ -1,6 +1,8 @@
 #ifndef SIMDUTF_UTF8_TO_UTF16_H
 #define SIMDUTF_UTF8_TO_UTF16_H
 
+#include <cstring>
+
 namespace simdutf {
 namespace scalar {
 namespace {

--- a/include/simdutf/scalar/utf8_to_utf16/valid_utf8_to_utf16.h
+++ b/include/simdutf/scalar/utf8_to_utf16/valid_utf8_to_utf16.h
@@ -1,6 +1,8 @@
 #ifndef SIMDUTF_VALID_UTF8_TO_UTF16_H
 #define SIMDUTF_VALID_UTF8_TO_UTF16_H
 
+#include <cstring>
+
 namespace simdutf {
 namespace scalar {
 namespace {

--- a/include/simdutf/scalar/utf8_to_utf32/utf8_to_utf32.h
+++ b/include/simdutf/scalar/utf8_to_utf32/utf8_to_utf32.h
@@ -1,6 +1,8 @@
 #ifndef SIMDUTF_UTF8_TO_UTF32_H
 #define SIMDUTF_UTF8_TO_UTF32_H
 
+#include <cstring>
+
 namespace simdutf {
 namespace scalar {
 namespace {

--- a/include/simdutf/scalar/utf8_to_utf32/valid_utf8_to_utf32.h
+++ b/include/simdutf/scalar/utf8_to_utf32/valid_utf8_to_utf32.h
@@ -1,6 +1,8 @@
 #ifndef SIMDUTF_VALID_UTF8_TO_UTF32_H
 #define SIMDUTF_VALID_UTF8_TO_UTF32_H
 
+#include <cstring>
+
 namespace simdutf {
 namespace scalar {
 namespace {

